app/validation/validator.py

```diff
@@ -179,15 +181,16 @@ def validate_routing_on_answer_options(self, block, answer, answer_ids):
         return answer_errors
 
     @staticmethod
-    def validate_when_rule(when_clause, existing_ids):
+    def validate_when_rule(when_clause, existing_ids, referenced_id):
         # Validates any answer id in a when clause exists within the schema
-        when_clause_ids = [when_clause[0]['id']]
+        for when in when_clause:
+            if 'id' in when:
+                message = 'The answer id - {} in the "when" clause for {} does not exist'.format(when['id'],
```

Comment:
Nitpick - consider building the `message` inside the following `if` condition since it's only relevant within that scope.